packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/pipeline-editor.spec.tsx

```diff
@@ -23,6 +23,9 @@ const renderPipelineEditor = (
         serverVersion="4.2"
         onChangePipelineText={() => {}}
         num_stages={1}
+        isPipelineLoadedFromExternal={false}
+        pipelineLoadedFromExternalKey={0}
```

Comment:
When `pipelineLoadedFromExternalKey` is `0`, does it always mean that `isPipelineLoadedFromExternal` `false`? I am wondering if we need `isPipelineLoadedFromExternal` at all, or if we could get the same usage from just `pipelineLoadedFromExternalKey`.

Reply:
usual comment about naming, what does "PipelineLoadedFromExternal" means? is there a name that would relate to this feature better?

Also more in general, i would have expected a bit less code, especially a bit less, and less spread around code for this feedback feature. Are we sure is worth?

Reply:
I've renamed it to `LoadedFromExternalSourceId`, how do you feel about that?

I removed the extra boolean, and hopefully cleaned up the interfaces a bit and reduces some of the code overhead.

I can't really think of a cleaner way apply an effect like this. We have to keep a flag around outside of rendering and ensure we clear it. We could do something hacky with applying a css class name when this happens directly from where it happens but I think that could be worse and coupled.

In terms of if it's worth it I'm not sure. I can bring it up with design, it does feel like a small visual impact that other things like the animated border and disabled inputs now assist in. Do you feel with the recent code changes this is unwieldily and not something we want to maintain?

Reply:
Maybe Maurizio was asking not necessarily about the `id` ending, for me it is not exactly clear what `LoadedFromExternalSource` means. Would we apply the same styles in the future when we load a query from the saved queries? Could you please also point out why we need this id? Because we can have several queries open on multiple tabs and we need to know which one we are working with?

Reply:
yeah, i have questions about "what is a source here", and what is an external source, is not really a concept we talk about outside of code, so i'm not sure if there is some naming that would speak more domain language, like `isPiplelineAIGenerated`, if that is what we are trying to track here.

Reply:
> Do we apply the same styles when we load a query from the saved queries
Yes
> Could you please also point out why we need this id?
We need this id in order for the frontend to know a new query has been applied. We need the effect to re-apply if a user run the same prompt twice and gets the same response back. A boolean could work but it would introduce more roundtrips in the component life cycle and each component that listens to the boolean would be attempting to un-set it once it has been set.

Reply:
> `isPipelineAIGenerated`
The effect is also shown when we load a pipeline from saved pipelines or query history. It's not something we need to do, we could not show it for those events and then we could use a naming like this. I was thinking something more generic in order to cover those extra use cases.

Reply:
Maybe something like `appliedChangesVersion`?